mantidproject/mantid
# Clang-tidy modernize-loop-convert MantidPlot

**Description of work.**
This PR runs clang-tidy on MantidPlot and docs (+ one qt file has snuck in here) to use range-based for loops where possible

**To test:**
1. Code inspection
2. Tests pass

Note that if you search mantidplot code for `for \(.*\.begin\(\);.*\.end\(\);.*\)` (regex for an old for loop) you will find some instances which remain. There are some instances where clang-tidy has decided they should not be / it doesn't know how to change the loops to range-based. These stragglers should be manually assessed at a later date

Fixes partially #22183 

*This does not require release notes* because **internal code change**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
